src/XrdSciTokens/XrdSciTokensAccess.cc

```diff
@@ -347,13 +359,67 @@ class OverrideINIReader: public INIReader {
 
 };
 
+// Given a list of access rules, this class determines whether a requested operation / path
+// is permitted by the access rules.
+class SubpathMatch final {
+public:
+    SubpathMatch() = default;
+    SubpathMatch(const AccessRulesRaw &rules)
+    : m_rules(rules)
+    {}
+
+    // Determine whether the known access rules permit the requested `oper` on `path`.
+    bool apply(Access_Operation oper, const std::string_view path) const {
+        auto is_subdirectory = [](const std::string_view& dir, const std::string_view& subdir) {
+            if (subdir.size() < dir.size())
+                return false;
+
+            if (subdir.compare(0, dir.size(), dir, 0, dir.size()) != 0)
+                return false;
+
+            return dir.size() == subdir.size() || subdir[dir.size()] == '/' || dir == "/";
+        };
```

Comment:
This is the function I mentioned, we can take it from `XrdOucPrivateUtils.hh` now (maybe keeping the code for this in the .cc file, as I don't see the advantage of moving it to the headers.